components/server/src/iam/iam-session-app.ts

```diff
@@ -31,34 +32,44 @@ export class IamSessionApp {
 
         app.post("/session", async (req: express.Request, res: express.Response) => {
             try {
-                await this.doCreateSession(req);
-                res.status(200).json();
+                const result = await this.doCreateSession(req);
+                res.status(200).json(result);
             } catch (error) {
-                res.status(500).json({ error, message: error.message });
+                // we treat all errors as bad request here and forward the error message to the caller
+                res.status(400).json({ error, message: error.message });
             }
         });
 
         return app;
     }
 
     protected async doCreateSession(req: express.Request) {
-        // We need an account to sign in, which is done by calling `req.login` below.
-        // As proper account creation/selection will be added in later on, we're creating a
-        // dummy account on each attempt.
-        //
-        const user = await this.userService.createUser({
-            identity: {
-                authId: "fake-id-" + Date.now(),
-                authName: "FakeUser",
-                authProviderId: "oidc1",
-                primaryEmail: "fake@email.io",
-            },
-            userUpdate: (user) => {
-                user.name = "FakeUser";
-                user.fullName = "Fake User";
-                user.avatarUrl = "https://github.com/github.png";
+        if (!OIDCCreateSessionPayload.is(req.body)) {
```

Comment:
This validation check just checks that the fields are present, but doesn't validate the values. Do we also need to validate these?

Reply:
Validation against what exactly? 
That's the ID token from an OIDC provider, which is validated in and pass on by the `iam` [internally](https://github.com/gitpod-io/gitpod/blob/243207eb85e1ac9ada023fc6ba3a4e3118d49d5e/components/iam/pkg/oidc/router.go#L89-L102).

Reply:
Sure. But every service should perform at least basic sanity checks on the fields. If nothing else, we should check that the fields are not empty. This acts as a safeguard if we get it wrong on the other service

Reply:
I mean, we need to rely on external services to adhere to their own contracts.
But I get the idea that - because this is the external boundary of "our" system - we want to be super-clear that we provide good, stable guarantees to all other "internal" components.

Reply:
Done [here](https://github.com/gitpod-io/gitpod/compare/a4035031234857f90d9f6f96171aafe10ae01a94..0d5c0d6dc02e5d4bca61482fe7baf3dd5afd587e)